vllm/spec_decode/proposer_worker_base.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional, Tuple
+
+from vllm.sequence import ExecuteModelRequest, SamplerOutput
+from vllm.spec_decode.interfaces import SpeculativeProposer
+from vllm.worker.worker_base import WorkerBase
+
+
+class ProposerWorkerBase(WorkerBase, SpeculativeProposer):
+    """Interface for proposer workers"""
+
+    @abstractmethod
+    def sampler_output(
+        self,
+        execute_model_req: ExecuteModelRequest,
+        sample_len: int,
+    ) -> Tuple[Optional[List[SamplerOutput]], bool]:
+        raise NotImplementedError
+
+    def set_include_gpu_probs_tensor(self):
+        """Implementation optional"""
+        pass
+
+
+class NonLLMProposerWorkerBase(ProposerWorkerBase, ABC):
```

Comment:
Extracted this since it would be common to other proposers that don't have their own kvcache (including MLPSpeculator)

Reply:
nit: Can we rename to `CachelessProposerWorkerBase`? Want to reserve the right for an LLM to not have a context cache

Reply:
Sure though for both these cases so far (Ngram and MLPSpeculator) `execute_model` is also a no-op, do you think that also fits?

Reply:
Sure, I'm also not sure what the most appropriate name would be. Revisiting sounds good!